src/ui/components/questions/questionsubmissioncomponent.js

```diff
@@ -323,10 +332,12 @@ export default class QuestionSubmissionComponent extends Component {
         questionDescription: formData.questionDescription
       })
         .catch(error => {
+          let errorMessage = this._config.networkErrorText;
+          if (error.code === 22) {
+            errorMessage = this._config.emailInvalidErrorText;
+          }
           this.setState(
-            new QuestionSubmission(formData, {
-              network: 'We\'re sorry, an error occurred.'
-            })
+            new QuestionSubmission(formData, { network: errorMessage })
```

Comment:
Setting network to errorMessage seems a little odd to me. How about `{ errorMessage: errorMessage }` instead?

Reply:
I think keeping the error name as network will be a bit more specific and can differentiate the backend network errors from the error check we have in the frontend for QuestionSubmission (nameError, questionPolicyError, questionTextError, etc.)